RecoLocalTracker/SiPixelRecHits/interface/PixelCPEFastParamsDevice.h

```diff
@@ -0,0 +1,32 @@
+#ifndef DataFormats_PixelCPEFastParams_interface_PixelCPEFastParamsDevice_h
+#define DataFormats_PixelCPEFastParams_interface_PixelCPEFastParamsDevice_h
+
+#include <alpaka/alpaka.hpp>
+#include "HeterogeneousCore/AlpakaInterface/interface/config.h"
+#include "HeterogeneousCore/AlpakaInterface/interface/memory.h"
+#include "FWCore/ParameterSet/interface/ParameterSet.h"
+#include "RecoLocalTracker/SiPixelRecHits/interface/PixelCPEGenericBase.h"
+
+#include "pixelCPEforDevice.h"
+
+template <typename TDev, typename TrackerTraits>
+class PixelCPEFastParamsDevice {
```

Comment:
After poking around in the code, I think we should not try to use `PortableDeviceObject` here, because the host and device types are significantly different:
  - the device type is pretty simple
  - the host type inherits from `PixelCPEGenericBase` that inherits from `PixelCPEBase`, and has a rather lengthy implementation of `fillParamsForDevice()`

Reply:
I guess a different solution could be to use a simpler object both on the host and on the device, and move the complex code to an ESProducer.
But, given the hierarchy of classes in the PixelCPE, I have the impression this would actually make the whole code more complicated.

Reply:
I was just wondering because this class template seems to replicate parts of the code/behavior of `PortableDeviceObject`, without really adding anything that isn't already in `PortableDeviceObject`.

If kept, I think this class should be made non-copyable (as `PortableDeviceObject` is). Otherwise it would be too easy to get a non-const access from a const object by making a copy (even if requires accessing the `buffer()`).

Reply:
Sure, I'll make this change.